src/AppInstallerCLITests/YamlManifest.cpp

```diff
@@ -862,6 +865,28 @@ TEST_CASE("ValidateV1_5GoodManifestAndVerifyContents", "[ManifestValidation]")
     VerifyV1ManifestContent(mergedManifest, false, ManifestVer{ s_ManifestVersionV1_5 });
 }
 
+TEST_CASE("WriteV1_5SingletonManifestAndVerifyContents", "[ManifestCreation]")
+{
+    ManifestValidateOption validateOption;
+    validateOption.FullValidation = true;
+    TempDirectory singletonDirectory{ "SingletonManifest" };
+    CopyTestDataFilesToFolder({ "ManifestV1_5-Singleton.yaml" }, singletonDirectory);
+    Manifest singletonManifest = YamlParser::CreateFromPath(singletonDirectory, validateOption);
+
+    // Override the exported manifest type to 'singleton' since validation does not accept 'merged' manifests.
+    auto setExportedManifestType = TestHook::SetExportedManifestTypeResult_Override(ManifestTypeEnum::Singleton);
+
+    TempDirectory downloadDirectory{ "downloadDirectory" };
+    std::filesystem::path testManifestPath = downloadDirectory.GetPath() / "testManifest.yaml";
+    const auto& manifestYamlContent = YamlWriter::ManifestToYamlString(singletonManifest, singletonManifest.Installers[0]);
+    YamlWriter::OutputYamlFile(manifestYamlContent, testManifestPath);
+
+    REQUIRE(std::filesystem::exists(testManifestPath));
+
+    Manifest generatedManifest = YamlParser::CreateFromPath(downloadDirectory, validateOption);
+    VerifyV1ManifestContent(generatedManifest, true, ManifestVer{ s_ManifestVersionV1_5 });

```

Comment:
Test a multifile scenario as well (multi installers and multi locales)

Reply:
Added tests for multifile and singleton scenario for all versions v1.0 - 1.6